dynamic_instrumentation/src/point/build.rs

```diff
@@ -24,19 +24,20 @@ struct InstrumentationPointBuilder<'tcx> {
     pub metadata: EventMetadata,
 }
 
-impl<'tcx> InstrumentationAdder<'_, 'tcx> {
-    fn add(&mut self, point: InstrumentationPointBuilder<'tcx>, loc: Location, func: DefId) {
+impl<'a, 'tcx> InstrumentationAdder<'a, 'tcx> {
+    fn add(&mut self, builder: InstrumentationBuilder<'a, 'tcx>) {
```

Comment:
I don't think `add` should take a whole `InstrumentationBuilder`, as that passes extra things that `add` doesn't need, like some fields and a lifetime.  The lifetime isn't an issue now, but I think minimizing lifetimes where possible is generally very useful, and once things get hairy with lifetimes, it gets hard.  Instead, we can just change the `loc: Location` arg to `original_location: Location, instrumentation_location: Location`.